### DISCUSSION TITLE
Filtro per stato fattura elettronica

### DISCUSSION THREAD

**Discussion by f107** [fattureincloud/api#235] [Q&A] [answered]
Ciao a tutti,

Stavo cercando un modo per filtrare la lista delle fatture per ricevere solo quelle ancora non inviate allo SDI, per poi inviarle in un secondo momento.

Da interfaccia FIC è possibile fare questo tipo di filtro, ma tramite API non ho trovato come fare, c'è un modo per farlo che mi sfugge?

Altrimenti sto valutando altre opzioni:
- prendere per ogni fattura nel range di tempo interessato lo stato.
- inviare le fatture nel range di tempo interessato e se una è già stata inviata gestire l'eccezione che l'API dovrebbe darmi.

Grazie

**Comment by emajo** [comment, marked as answer]
Ciao @f107 purtroppo attualmente il campo e_invoice non è filtrabile, una soluzione sarebbe si quella di fare un list degli issued documents con filtro sul range di date interessate, e poi inviare solo le fatture elettroniche (e_invoice=true) con status=not_sent, così da non dover gestire gli errori per i documenti già inviati.

Grazie,
Emanuele.

**Comment by f107** [comment]
Ok, giusto per vedere se ho capito, uso il metodo lista per il range di data interessato, con `fieldset: detailed` per ricevere anche il campo ei_status e mi prendo solo quelle `e_invoice = true` ed` ei_status = not_sent` ?

Grazie

**Reply by emajo** [reply]
Esattamente
